compiler/rustc_typeck/src/check/fn_ctxt/checks.rs

```diff
@@ -296,7 +336,26 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             } else {
                 false
             };
-            param_count_error(expected_arg_count, supplied_arg_count, "E0061", false, sugg_unit);
+
+            // are we passing elements of a tuple without the tuple parentheses?
+            let expected_input_tys = if expected_input_tys.is_empty() {
+                // In most cases we can use expected_input_tys, but some callers won't have the type
+                // information, in which case we fall back to the types from the input expressions.
+                formal_input_tys
+            } else {
+                &*expected_input_tys
+            };
+
+            let sugg_tuple_wrap_args = self.suggested_tuple_wrap(expected_input_tys, provided_args);
```

Comment:
Can you pass `supplied_types` instead to avoid re-checking the expressions?

Reply:
I'm not sure I follow, I can't see any potential double `self.check()` calls in the caller (except the `...` arguments to a variadic functions), and no loops - maybe I've got the wrong end of the stick?

Reply:
Hmm, I think I must have gotten confused. Sorry about that.

Reply:
No worries :)